packages/edit-site/src/components/header/fullscreen-mode-close/index.js

```diff
@@ -15,7 +26,8 @@ function FullscreenModeClose( { icon } ) {
 		return null;
 	}
 
-	const buttonIcon = icon || wordpress;
+	const shouldDisplaySiteIcon = siteIconURL || isRequestingSiteIcon;
+	const buttonIcon = shouldDisplaySiteIcon ? null : icon || wordpress;
```

Comment:
TBH I don't know the order of precedence for a ternary vs a `||`. Can we rewrite this a bit to make it clearer?

Reply:
I totally agree 👍 . This is something I wanted to clean up (along with styling) before a final code review.

Reply:
I clean up the conditional logic [here](https://github.com/WordPress/gutenberg/pull/22952/commits/1b23ed352d612ad0d235d973c0c1b5fd27bd17d5). Let me know if that's easier to read.